data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -87,21 +89,26 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
             keyValueDelimiterPattern = Pattern.compile(regex);
         }
 
-        if(!validateRegex(keyValueProcessorConfig.getDeleteKeyRegex())) {
+        if (!validateRegex(keyValueProcessorConfig.getDeleteKeyRegex())) {
             throw new PatternSyntaxException("delete_key_regex is not a valid regex string", keyValueProcessorConfig.getDeleteKeyRegex(), -1);
         }
 
-        if(!validateRegex(keyValueProcessorConfig.getDeleteValueRegex())) {
+        if (!validateRegex(keyValueProcessorConfig.getDeleteValueRegex())) {
             throw new PatternSyntaxException("delete_value_regex is not a valid regex string", keyValueProcessorConfig.getDeleteValueRegex(), -1);
         }
 
-        if(keyValueProcessorConfig.getIncludeKeys() != null) {
+        if (keyValueProcessorConfig.getIncludeKeys() != null) {
             includeKeysSet.addAll(keyValueProcessorConfig.getIncludeKeys());
         }
 
-        if(!validTransformOptionSet.contains(keyValueProcessorConfig.getTransformKey())) {
+        if (!validTransformOptionSet.contains(keyValueProcessorConfig.getTransformKey())) {
             throw new IllegalArgumentException(String.format("The transform_key value: %s is not a valid option", keyValueProcessorConfig.getTransformKey()));
         }
+
+        if (!(keyValueProcessorConfig.getWhitespace().equals(WHITESPACE_STRICT)
```

Comment:
if whitespace is lenient, basically no modification needs to be done to the key and value, so I just excluded it from the logic. should I still use a set in this case?

Reply:
Since here it is just validating the value of whitespace parameter, you could use a set of valid values to simplify the condition statement logic
```suggestion
        if (WHITESPACE_VALID_VALUES.contains(keyValueProcessorConfig.getWhitespace()))
```

Reply:
ah, I see the full context now. changing and will commit soon